src/components/settings/SettingsLMSTab/LMSConfigs/utils.tsx

```diff
@@ -0,0 +1,177 @@
+import handleErrors from "../../utils";
+import { camelCaseDict } from "../../../../utils";
+// @ts-ignore
+import { setWorkflowStateAction, updateFormFieldsAction } from "../../../forms/data/actions.ts";
+import { CanvasConfigCamelCase, CanvasConfigSnakeCase } from "./Canvas/CanvasConfig";
+import { BlackboardConfigCamelCase, BlackboardConfigSnakeCase } from "./Blackboard/BlackboardConfig";
+import { CornerstoneConfigCamelCase, CornerstoneConfigSnakeCase } from "./Cornerstone/CornerstoneConfig";
+import { DegreedConfigCamelCase, DegreedConfigSnakeCase } from "./Degreed/DegreedConfig";
+import { MoodleConfigCamelCase, MoodleConfigSnakeCase } from "./Moodle/MoodleConfig";
+import { SAPConfigCamelCase, SAPConfigSnakeCase } from "./SAP/SAPConfig";
+// @ts-ignore
+import { FormWorkflowErrorHandler, WAITING_FOR_ASYNC_OPERATION } from "../../../forms/FormWorkflow.tsx";
+
+type ConfigCamelCase =
+  BlackboardConfigCamelCase | CanvasConfigCamelCase | CornerstoneConfigCamelCase | DegreedConfigCamelCase | MoodleConfigCamelCase | SAPConfigCamelCase;
+type ConfigSnakeCase =
+  BlackboardConfigSnakeCase | CanvasConfigSnakeCase | CornerstoneConfigSnakeCase | DegreedConfigSnakeCase | MoodleConfigSnakeCase | SAPConfigSnakeCase;
+
+export const LMS_AUTHORIZATION_FAILED = "LMS AUTHORIZATION FAILED";
+
+export async function handleSubmitHelper(
+  enterpriseCustomerUuid: string,
+  transformedConfig: ConfigSnakeCase,
+  existingData: any,
+  onSubmit: Function,
+  formFieldsChanged: Boolean,
+  currentFormFields: any,
+  lmsType: string,
+  channelMap: { [key: string]: {[key: string]: any }},
+  errHandler: FormWorkflowErrorHandler | undefined,
+  dispatch: any,
+) {
+  transformedConfig.enterprise_customer = enterpriseCustomerUuid;
+  let err = "";
+  if (formFieldsChanged) {
+    if (currentFormFields?.id) {
+      try {
+        transformedConfig.active = existingData.active;
+        const response = await channelMap[lmsType].update(transformedConfig, existingData.id)
+        currentFormFields = camelCaseDict(
+          response.data
+        ) as ConfigCamelCase;
+        onSubmit(currentFormFields);
+        dispatch?.(updateFormFieldsAction({ formFields: currentFormFields }));
+      } catch (error) {
+        err = handleErrors(error);
+      }
+    } else {
+      try {
+        transformedConfig.active = false;
+        const response = await channelMap[lmsType].post(transformedConfig)
+        currentFormFields = camelCaseDict(
+          response.data
+        ) as ConfigCamelCase;
+        onSubmit(currentFormFields);
+        dispatch?.(updateFormFieldsAction({ formFields: currentFormFields }));
+      } catch (error) {
+        err = handleErrors(error);
+      }
+    }
+  }
+  if (err) {
+    errHandler?.(err);
+  }
+  if ((lmsType === BLACKBOARD_TYPE || lmsType === CANVAS_TYPE) && currentFormFields && !currentFormFields?.refreshToken) {
```

Comment:
There's a lot of branching logic here.  Would it make sense to have separate helper methods for Blackboard and Canvas?